src/Dhall.hs

```diff
@@ -344,6 +344,14 @@ text = Type {..}
 
     expected = Text
 
+{-| Decode strict `Text`
+
+>>> input text' "\"Test\""
+"Test"
+-}
+text' :: Type Data.Text.Text
```

Comment:
Minor suggestion: perhaps call this `strictText` and rename the old one to `lazyText`.  It's okay if it's a breaking change